### PR TITLE
Teacher Calendar, foundation

### PR DESCRIPTION
**Pull request by pandafulmanda**
- [x] pull out plan on calendar as it's own component with it's own behaviors, should be usable in week view as well
- [ ] abstract out data reshaping/filtering code in a separate helper
  - [ ] not quite pleased with where this is, will likely address when connecting with backend.
- [x] improve calendar view controller navigation
- [x] clean up styles, make betttttterrrrr
- [x] test tests testsssss
  - [x] should render calendar
  - [x] should be able to navigate months on calendar
  - [x] should render plans on calendar
  - [x] should trigger event on plan click
- [x] figure out failed builds
  - that was my fault yayyy

screen shot 2015-04-16 at 10 25 25 am

Install instructions (since it uses a non-npm-packaged version):

```sh
cd node_modules/react-calendar/
npm install
npm run prepublish
```

### CODE REVIEW

**Comment by pandafulmanda** [comment]
wont really work until the PR is merged.

File: src/components/course-calendar/course-month.cjsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    )
+
+  render: ->
+
+    # TODO see about doing a PR to the library after their react update to enable
+    # modifying MonthHeader as a feature
+    MonthHeader =
+      <BS.Row className='month-header'>
+        <BS.Col xs={4}>
+          <a href="#" className='month-header-control previous' onClick={@handlePreviousMonth}><</a>
```
Comment (philschatz) [inline src/components/course-calendar/course-month.cjsx]: Does `&lt;` work (unsure how react handles escaping)? It just looks a bit odd

File: src/components/course-calendar/course-month.cjsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+          <a href="#" className='month-header-control next' onClick={@handleNextMonth}>></a>
+        </BS.Col>
+      </BS.Row>
+
+
+    <BS.Grid>
+      {MonthHeader}
+      <BS.Row>
+        <BS.Col xs={12}>
+          <Month date={@state.date}/>
```
Comment (philschatz) [inline src/components/course-calendar/course-month.cjsx]: Are you able to render links for each day? Maybe something like:

```coffee
<Month date={@state.date}>
  <Day date={...}><div>HW4</div></Day>
  <Day date={...}><div>HW5</div></Day>
</Month>
```

It is unclear to me from the Readme if that is possible.

File: style/calendar.less [hunk 3]
```diff
@@ -0,0 +1,125 @@
+// Sample bootstrap theme for react-calendar
```
Comment (philschatz) [inline style/calendar.less]: Could we initially just `@import '../node_modules/react-calendar/...` instead of copying it into our codebase?

File: src/components/course-calendar/course-month.cjsx (hunk 2 above)
Comment (philschatz) [inline src/components/course-calendar/course-month.cjsx]: Ok, the following works (but changes the calendar styling a bit):

```coffee
    hw4 = <span onClick={-> alert('Clicked HW 4')}>HW4</span>

    <BS.Grid>
      {MonthHeader}
      <BS.Row>
        <BS.Col xs={12}>
          <Month date={@state.date}>
            <Day date={moment('2015-04-11')} dayAgenda={true} >{hw4}</Day>
          </Month>
        </BS.Col>
      </BS.Row>
    </BS.Grid>
```

image

File: style/calendar.less (hunk 3 above)
Comment (pandafulmanda) [inline style/calendar.less]: yes =)

Comment (pandafulmanda) [inline style/calendar.less]: actually, not without including some webpack dependencies or writing something for our gulp styles task to handle:

```less
@import '~bootstrap/less/bootstrap.less';
@import '~bootstrap/less/theme.less';
```

I will shelf this for now, and work on other aspects of the calendar first

File: src/components/course-calendar/course-month.cjsx (hunk 2 above)
Comment (philschatz) [inline src/components/course-calendar/course-month.cjsx]: Seems it may be easier to use `display: table` and `display: table-cell` for the calendar (just write CSS from scratch) instead of `display: inline-block` (which seems to be causing the mis-aligned dates)

Comment (pandafulmanda) [inline src/components/course-calendar/course-month.cjsx]: okay

**Comment by pandafulmanda** [comment]
- [ ] pull out plan on calendar as it's own component with it's own behaviors, should be usable in week view as well
- [ ] abstract out data reshaping/filtering code in a separate helper
- [ ] improve calendar view controller navigation
- [ ] clean up styles, make betttttterrrrr
- [ ] test tests testsssss

**Comment by philschatz** [comment]
Yay! How does it handle the corner cases:

- range from Feb 28 - Mar 1
- range from Mar 31 - Apr 1
- range from Mar 31 - Apr 5
- > 3 events on the same date

**Comment by pandafulmanda** [comment]
I'll add some events,  i think the first 3 cases will be fine, but the > 3 events...we could either increase the height of the boxes, or have a ... for showing more?
